plutus-contract/plutus-contract.cabal

```diff
@@ -83,6 +83,7 @@ library
         tasty -any,
         tasty-hunit -any,
         row-types -any,
+        freer-simple -any,
```

Comment:
I'd try [`fused-effects`](http://hackage.haskell.org/package/fused-effects) first, because it supports higher-order effects (not sure if this is good, though) and seems to be a bit more widely used. On the other hand if a simple thing works, why complicate your life...

Reply:
I thought `polysemy` is all the rage right now.

Reply:
`freer-simple` was the first one I could get to build: `polysemy` got stuck since most of its dependencies aren't in the stack snapshot we're using. I didn't try `fused-effects` since `freer-simple` seems to do most of what I want and I don't care about fancy stuff like higher-order effects, but I can look at it. Fortunately they all seem to use similar styles of defining effects.